Firmware/ultralcd.cpp

```diff
@@ -6929,29 +6941,61 @@ bool lcd_selftest()
 
 	_progress = lcd_selftest_screen(testScreen::extruderFan, _progress, 3, true, 2000);
 #if (defined(FANCHECK) && defined(TACH_0))
-	_result = lcd_selftest_fan_dialog(0);
+	switch (lcd_selftest_fan_auto(0)){		// check extruder Fan
+		case TestError::extruderFan:
+			_result = false;
+			break;
+		case TestError::swappedFan:
+			_swapped_fan = true;
```

Comment:
This may generate compiler warning or catch someones eye. You can suppress it by comment
// no break